src/main/java/com/crud_read_and_create/service/GameService.java

```diff
@@ -21,18 +26,31 @@ public GameService(GameMapper gameMapper) {
 		this.gameMapper = gameMapper;
 	}
 
-	public List<Game> getGamesAsc() {
-		return gameMapper.findAllAsc();
-	}
-
-	public List<Game> getGamesDesc() {
-		return gameMapper.findAllDesc();
-	}
+	public void getGames(GameForm gameForm, Model model) {
 
-	public List<Game> getGamesId(String id) {
-		Game gameId = new Game();
-		gameId.setId(id);
-		return gameMapper.findById(gameId);
+		if (StringUtils.isEmpty(gameForm.getId())) {
+			if (gameForm.getOrder().equals("asc")) {
+				List<Game> gameListAsc = gameMapper.findAllAsc();
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg)
別PRでよいですがasc/descをMapperのfindAllの引数に渡すのもありだなあと思いました。

List<Game> findAll(OrderBy orderBy);
OrderByというEnumを作ってASCとDESCを定義するイメージ。

ここわかりにくかったらoviceか何かで解説します。

Reply:
https://github.com/renangton/task5_crud_read_and_create/pull/18/commits/9b5bb9774bdc9ed93af92cbb435cf8c1cfdd2d52

MapperのfindAllの引数にASCとDESCを渡し、GameMapper.xmlの方で分岐させることには成功したのですが、分岐のさせ方が自分のやりたいようにはできませんでした。

## やりたかったこと
```
<choose>
	<when test="orderBy=orderBy.name() == 'ASC'">
		ORDER BY games_platforms.id
		ASC
	</when>
	<when test="orderBy=orderBy.name() == 'DESC'">
		ORDER BY games_platforms.id
		DESC
	</when>
	<otherwise>
		ORDER BY games_platforms.id
	</otherwise>
</choose>
```

## エラー内容
上記コード時にASCを選択している時は、問題なく処理が実行できるのですが、DESC選択時は下記のエラーが出ました。
ASCとDESCの順番を入れ替えた時は、DESCは成功し、ASCは同様のエラーが出ました。
```
There was an unexpected error (type=Internal Server Error, status=500).
nested exception is org.apache.ibatis.builder.BuilderException: Error evaluating expression 'orderBy=orderBy.name() == 'DESC''. Cause: org.apache.ibatis.ognl.MethodFailedException: Method "name" failed for object false [java.lang.NoSuchMethodException: java.lang.Boolean.name()]
org.mybatis.spring.MyBatisSystemException: nested exception is org.apache.ibatis.builder.BuilderException: Error evaluating expression 'orderBy=orderBy.name() == 'DESC''. Cause: org.apache.ibatis.ognl.MethodFailedException: Method "name" failed for object false [java.lang.NoSuchMethodException: java.lang.Boolean.name()]
```

##考察
状況的に初回のnameメソッドで値を取得することができ、２回目以降はメソッドを実行できないということは分かったのですが、エラー文を調査してみても、２回目以降が実行できない原因は分かりませんでした。

## 参考資料
https://mybatis.org/mybatis-3/ja/dynamic-sql.html
https://stackoverflow.com/questions/12933813/using-enum-parameters-in-mybatis-dynamic-sql

Reply:
こちらは私の方でも動作確認してみます。

Reply:
https://github.com/renangton/task5_crud_read_and_create/pull/18/commits/d70f2c91399c463a4a48eca7863061c88401864e
対応しました！

## メモ
${　}・・・中身をそのまま出力
#{　}・・・エスケープや文字列の場合、シングルクォーテーションで囲んで出力
$だと中身をそのまま出力するため、意図しない内容を送られて、SQLインジェクションをされる可能性あり。$を使用する際はSQLインジェクションを考慮する。

## 参考資料
https://www.codetd.com/ja/article/12615573
https://note.com/sudatakashi/n/n66f522f9ba7c

Reply:
https://github.com/renangton/task5_crud_read_and_create/pull/18/commits/ba82adcfe172746f7cd2f95eb8d8f97d651d2db1
OrderByにfromメソッドを追加し、Serviceでの分岐条件をなくしました。

## メモ
```
Arrays.stream(values())
.filter(v -> v.name().equals(StringUtils.upperCase(input)))
.findFirst().orElseThrow(RuntimeException::new);
```
values()でEnumで定義したASCとDESCをListで取得
→戻り値がListなので、Arrays.stream()
v.name()で取得したListから値を1個だけ取り出し、equals()で比較
StringUtils.upperCase(input)でgameForm.getOrder()の戻り値を大文字に変換
findFirst()で最初に一致した値を返す
一致するものがなかった時に、orElseThrow(RuntimeException::new)で
ランタイムエラーを出す